File: PHP/Behavioral/ChainOfResponsibility/Example2/CatHandler.php
<?php


namespace TheRayCode\ChainOfResponsibility;


class CatHandler extends AbstractHandler
{
    public function handle(string $request): ?string
    {
        if ($request === "Catnip") {
            return "Cat: I'll eat the " . $request . ".<br/>";
        } else {
            return parent::handle($request);
        }
    }
}

File: TheRayCode/CSharp/Behavioral/ChainOfResponsibility/Example/ChainOfResponsibility/CatHandler.cs
namespace ChainOfResponsibility
{
    class CatHandler : AbstractHandler
    {
        public override object Handle(object request)
        {
            if (request.ToString() == "Catnip")
            {
                return $"Cat: I love {request.ToString()}.\n";
            }
            else
            {
                return base.Handle(request);
            }
        }
    }
}